packages/website/content/docs/overview/getting-started.md

```diff
@@ -42,47 +46,82 @@ In the same `src/App.tsx` file, use `InefficientComponent`:
       </InefficientComponent>
 ```
 
-4. Modify Build Configuration
+### 4. Modify build configuration
 
-In the app's root directory, modify `webpack.config.js`:
+In order to generate the flamegraphs for the app, we need to build the app for production. Normally, we'd simply
+run `yarn build` which is an alias for `react-scripts build`, which in turn, under-the-hood calls `webpack` to
+optimize and minify the app. However, for the flamegraphs to be readable and meaningful, we need to disable the
+minification. While `react-scripts` don't allow us to tweak the `webpack` out of the box, it is possible via
+`react-app-rewired` package. With `react-app-rewired` added as a dev dependency, we can tell it to override certain
+configuration values of `webpack` such as turning minification off for the production build.
 
-```tsx
-  {
-    // Here you can add custom webpack configurations
-    optimization: {
-      // We no not want to minimize our code.
-      minimize: false
-    },
+#### 4.1 Add `react-app-rewired` dependency:
+
+`yarn add react-app-rewired --dev`
+
+#### 4.2 Tweak `package.json` to use `react-app-rewired`
+
+```json
+// ...
+"scripts": {
+  "build": "react-app-rewired build"
+}
+```
+
+#### 4.3 Create build config override:
+
+`touch config-overrides.js`
+
+And override the `webpack` minification config option:
+
+```js
+module.exports = {
+  webpack: function(config, env) {
+    if (env === 'production') {
+      config.optimization.minimize = false;
+    }
+    return config;
   }
+}
 ```
 
-5. Build App
+### 5. Build your app
 
 `yarn build`
 
-This should generate `dist/index.html` which we will use in flamegrill.
+This should put your optimized app in `build/` which we will serve and analyze with Flamegrill.
 
+### 6. Serve your app
+
+Finally, serve your app, e.g., with `serve`:
+
+`serve -s build`
 
 ## Run Flamegrill
 
-1. [Install Flamegrill](./installation)
+### 1. [Install Flamegrill](./installation)
+
+### 2. Create an output directory
 
-2. Create an Output Directory
+Flamegrill will create file output, so find or create a directory from which to run Flamegrill. For instance,
 
-Flamegrill will create file output, so find or create a directory from which to run flamegrill.
+`mkdir perf-results`
 
-3. Run Flamegrill
+### 3. Run Flamegrill
 
-Modifying the path as appropriate, run flamegrill against `dist/index.html` generated with the build above:
+Modifying the path as appropriate, run Flamegrill against the URL where you serve your app with `serve`. By default,
+this would be `localhost:5000`:
```

Comment:
I tend to think using a static file is better here. It shows that any arbitrary file locator works and that local dev server isn't actually needed (as is the case with various pipelines implementing flamegrill in CI.)

Reply:
So for some reason, I haven't got to the reason why yet, analysing static file will not work in this case -- the generated flamegraph comes back empty. Lemme have a closer look to figure out why. It seems to me that maybe the production assets generated with `react-scripts` and whatever version of `webpack` they wrap internally might require spawning a server instance to actually serve the site. But this is just a hunch, I'll dig in deeper and report back.

Reply:
OK, I figured it out. The reason nothing was loading when opening the `index.html` directly (or pointing flamegrill at it directly) was because `react-scripts` configures `webpack` to use absolute paths for all the assets outputted from the production build. I've now applied one additional tweak to the `webpack` config that's described in the guide and flamegrill works great by pointing it at the main website entrypoint.